nvc/toc.go

```diff
@@ -0,0 +1,82 @@
+package nvc
+
+import (
+	"fmt"
+	"io"
+	"errors"
+	"encoding/binary"
+)
+
+// Reads the Magic Header and Table of Contents from r.
+// The ToC format is as follows:
+//   1. Magic bytes header (technically not part of the ToC but it comes first)
+//   2. 32 LE unsigned integer -> number of entries in ToC.
+//   3. This repeated as many times as the last value indicates:
+//     a. FNV1a hash (64-bit LE unsigned integer)
+//     b. Offset in the NVC archive (64-bit LE unsigned integer)
+//     c. Uncompressed length of the Entry (64-bit LE unsigned integer)
+//     d. Actual length of the Entry (64-bit LE unsigned integer)
+//     e. Entry flags (64-bit LE unsigned integer)
+func ReadToc(r io.Reader) ([]TocEntry, error) {
+	if magicErr := readMagic(r) ; magicErr != nil {
+		return nil, magicErr
+	}
+	count, countErr := readCount(r)
+	if countErr != nil {
+		return nil, countErr
+	}
+
+	entries := make([]TocEntry, count)
+
+	for i := range(entries) {
```

Comment:
Nit: I think it would be better to do `for i:=0; i<count; i++` here rather than ranging over `entries`.